src/newgrf_config.h

```diff
@@ -83,6 +84,16 @@ struct GRFIdentifier {
 	uint32 grfid;     ///< GRF ID (defined by Action 0x08)
 	uint8 md5sum[16]; ///< MD5 checksum of file to distinguish files with the same GRF ID (eg. newer version of GRF)
 
+	GRFIdentifier() = default;
+	GRFIdentifier(const GRFIdentifier &other) = default;
+	GRFIdentifier(GRFIdentifier &&other) = default;
+	GRFIdentifier(uint32 grfid, const uint8 *_md5sum) : grfid(grfid)
```

Comment:
I don't like the leading underscore in `_md5sum`, it's fine to hide the member variable with a parameter named the same. The member is referenced via `this->` anyway.
Reserve leading underscores for globals.